Make time attr round-trippable

donaldong opened the pull request:
https://apidock.com/ruby/Time/nsec
> The lowest digits of #to_f and #nsec are different because IEEE 754 double is not accurate enough to represent the exact number of nanoseconds since the Epoch.

> The more accurate value is returned by #nsec.

This converts time from a float to integer so it can become round-trippable.

**This is a breaking change!** The existing records are not compatible with the new version.

### Test Plan
- [x] Modified test cases to cover this behavior

donaldong commented:
This is related to #72 as well

sibaranov commented on lib/redcord/serializer.rb:
```diff
@@ -31,7 +31,7 @@ module ClassMethods
     def encode_attr_value(attribute, val)
       if !val.blank? && TIME_TYPES.include?(props[attribute][:type])
-        val = val.to_f
+        val = val.to_i * 1_000_000_000 + val.nsec
```
What if `val.to_i` is negative? Do we need to raise an error or support such case?

donaldong replied:
hmm I'm not sure when this gonna happen when it's a TIME_TYPES  but happy to add a validation

sibaranov replied:
It can be when time is before year 1970.

donaldong commented:
support negative time

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/redcord/compare/7200f21963d698dfa26730fc6944d93366e71fcb..e303ae7782e8914ee2189f3c6b973b6cbc804aea).

sibaranov reviewed (approved):
looks good!
